### PR TITLE
Add max_allowed_packet diagnostic check

### LINKED ISSUE #1781: New diagnostic checking that max_allowed_packet is at least 64M or issue warning, and recommend 128M
From http://dev.mysql.com/doc/refman/5.1/en/packet-too-large.html

```
When a MySQL client or the mysqld server receives a packet bigger than max_allowed_packet bytes, it issues a Packet too large error and closes the connection. With some clients, you may also get a Lost connection to MySQL server during query error if the communication packet is too large.
```

Default server configuration is 1MB.  For larger sites, it needs to be increased to MEDIUMBLOB (16MB).

### PR DESCRIPTION
**Pull request by tsteur**
fix #1781 

thought about linking to https://matomo.org/faq/troubleshooting/faq_183/ but the FAQ is actually not very helpful.

### CODE REVIEW

File: plugins/Diagnostics/Diagnostic/DbMaxPacket.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        $status = DiagnosticResult::STATUS_OK;
+        $label = $this->translator->translate('Diagnostics_MysqlMaxPacketSize');
+        $comment = '';
+
+        $minSize = self::MIN_VALUE_MAX_PACKET_MB * 1000 * 1000; // not using 1024 just in case... this amount be good enough
+        if (!empty($maxPacketBytes['Value']) && $maxPacketBytes['Value'] < $minSize) {
+            $status = DiagnosticResult::STATUS_WARNING;
+            $pretty = MetricsFormatter::getPrettySizeFromBytes($maxPacketBytes['Value'], 'M');
+            $configured = str_replace(array(' M', '&nbsp;M'), 'MB', $pretty);
+            $comment = Piwik::translate('Diagnostics_MysqlMaxPacketSizeWarning', array('32MB', $configured));
```
Comment (fdellwing) [inline plugins/Diagnostics/Diagnostic/DbMaxPacket.php]: This should not be hardcoded I think.

Comment (sgiehl) [reply plugins/Diagnostics/Diagnostic/DbMaxPacket.php]: The referenced issue suggest to set it to at least 64MB, but guess 32MB might be fine as well?

**Review by sgiehl — APPROVED** [review]
Left a small comment. Otherwise looks good and worked locally.

File: plugins/Diagnostics/Diagnostic/DbMaxPacket.php (hunk 1 above)
Comment (tsteur) [reply plugins/Diagnostics/Diagnostic/DbMaxPacket.php]: I don't remember exactly but we talked about it in chat or so. Somewhere 32 was said. Can change it to 64.

**Pull request merged**